specifyweb/frontend/js_src/lib/components/WorkBench/WbView.tsx

```diff
@@ -49,13 +48,298 @@ import type { WbMapping } from './mapping';
+        spreadsheetChanged,
+        checkDeletedFail
+      ),
+    [dataset, hotIsReady]
+  );
+
+  const isMapped = Boolean(dataset.uploadplan);
+  const canUpdate = hasPermission('/workbench/dataset', 'update');
+
+  const [showToolkit, _openToolkit, _closeToolkit, toggleToolkit] =
+    useBooleanState();
+
+  // TODO: move all states and logic for Results into a hook
+  const [showResults, _, closeResults, toggleResults] = useBooleanState();
+
+  const initialHiddenRows = React.useMemo(() => {
+    if (!hotRef.current) return [];
+    else return getHotPlugin(hotRef.current.hotInstance!, 'hiddenRows').getHiddenRows();
+  }, [hotRef.current]);
+  const initialHiddenCols = React.useMemo(() => {
+    if (!hotRef.current) return [];
+    else return getHotPlugin(hotRef.current.hotInstance!, 'hiddenColumns').getHiddenColumns();
+  }, [hotRef.current]);
+
+  // Makes the hot changes required for upload view results
+  React.useEffect(() => {
+    if (!workbench.hot) return;
+
+    const rowsToInclude = new Set();
+    const colsToInclude = new Set();
+    Object.entries(workbench.cells.cellMeta).forEach(([physicalRow, rowMeta]) =>
+      rowMeta.forEach((metaArray, physicalCol) => {
+        if (!workbench.cells.getCellMetaFromArray(metaArray, 'isNew')) return;
+        rowsToInclude.add((physicalRow as unknown as number) | 0);
+        colsToInclude.add(physicalCol);
+      })
+    );
+    const rowsToHide = workbench.data
+      .map((_, physicalRow) => physicalRow)
+      .filter(
+        (physicalRow) =>
+          !rowsToInclude.has(physicalRow) &&
+          !initialHiddenRows.includes(physicalRow)
+      )
+      .map(workbench.hot.toVisualRow);
+    const colsToHide = workbench.dataset.columns
+      .map((_, physicalCol) => physicalCol)
+      .filter(
+        (physicalCol) =>
+          !colsToInclude.has(physicalCol) &&
+          !initialHiddenCols.includes(physicalCol)
+      )
+      .map(workbench.hot.toVisualColumn);
+
+    if (showResults) {
+      workbench.hot?.updateSettings({ readOnly: true });
+      getHotPlugin(workbench.hot, 'hiddenRows').hideRows(rowsToHide);
+      getHotPlugin(workbench.hot, 'hiddenColumns').hideColumns(colsToHide);
+
+      workbench.utils.toggleCellTypes(
+        'newCells',
+        'remove',
+        spreadsheetContainer.current
+      );
+    } else {
+      getHotPlugin(workbench.hot, 'hiddenRows').showRows(
+        rowsToHide.filter((visualRow) => !initialHiddenRows.includes(visualRow))
+      );
+      getHotPlugin(workbench.hot, 'hiddenColumns').showColumns(
+        colsToHide.filter((visualCol) => !initialHiddenCols.includes(visualCol))
+      );
+    }
+
+    workbench.hot.render();
+  }, [showResults]);
+
+  return (
+    <>
+      <div
+        className="flex items-center justify-between gap-x-1 gap-y-2 whitespace-nowrap"
+        role="toolbar"
+      >
+        <div className="contents">
+          <DataSetName
+            dataset={dataset}
+            getRowCount={() =>
```

Comment:
(optional) pass a `hot` instance to this component and have it define the `getRowCount` function inside of it?
no need to define it inside of this component as it's already somewhat large (and important - so shouldn't be bothered with trivial things like defining getRowCount callback 😊 - that should be responsibility of DataSetName)